### PR TITLE
Fix kubectl describe to display a container's resource limit env vars as node allocatable when the limits are not set

### PR DESCRIPTION
**Pull request by aveshagarwal**
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/29849)

### CODE REVIEW

**Comment by aveshagarwal** [comment]
@kubernetes/rh-cluster-infra

File: pkg/kubectl/describe.go [hunk 1]
```diff
@@ -927,6 +935,9 @@ func describeContainers(label string, containers []api.Container, containerStatu
 				}
 				fmt.Fprintf(out, "      %s:\t%s (%s:%s)\n", e.Name, valueFrom, e.ValueFrom.FieldRef.APIVersion, e.ValueFrom.FieldRef.FieldPath)
 			case e.ValueFrom.ResourceFieldRef != nil:
+				if nodeCapacity != nil {
+					fieldpath.MergeContainerResourceLimitsWithCapacity(&container, nodeCapacity)
```
Comment (sttts) [inline pkg/kubectl/describe.go]: Do I understand this correctly that for a 4 GB node I will see a 4GB limit in the pod output as limit? Wouldn't it make sense to print that more explicitly? Something like:

Memory limits: unlimited, up to node capacity 4GB

Comment (aveshagarwal) [inline pkg/kubectl/describe.go]: @sttts Your understanding is correct. 
Regarding changing output string, lets see if others have the same opinion and then I would be happy to update it. But in my view, I think it is pretty clear that any value there is upper bound anyway.

Comment (sttts) [inline pkg/kubectl/describe.go]: I think my main point is that having no limit is not good. Printing "unlimited" would transport this message. Exhausting the node capacity can have any unexpected result.

**Comment by aveshagarwal** [comment]
Fixed unit tests.

**Comment by derekwaynecarr** [comment]
@aveshagarwal - can you paste a sample pod I can use to test this?

**Comment by aveshagarwal** [comment]
@derekwaynecarr https://bugzilla.redhat.com/show_bug.cgi?id=1357737 has one.

**Comment by ncdc** [comment]
This is not going to work for OpenShift because the majority of users won't be allowed to retrieve information about the node. We're still debating if any action is required for the bz. If not, we'll close this PR.

**Comment by aveshagarwal** [comment]
@ncdc what is the concern in trying best effort? That means if node capacity can not be accessed for whatever reason, continue with 0 if limits are not set. And display node capacity if its available and can be accessed if limits are not set.

**Comment by ncdc** [comment]
Because it will fail the majority of the time for most users once sensible RBAC settings are in place

File: pkg/kubectl/describe.go [hunk 2]
```diff
@@ -501,10 +501,18 @@ func (d *PodDescriber) Describe(namespace, name string, describerSettings Descri
 		}
 	}
 
-	return describePod(pod, events)
+	var nodeCapacity api.ResourceList
+	node, err := d.Nodes().Get(pod.Spec.NodeName)
+	if err != nil {
+		glog.Errorf("unable to retrieve node capacity for resource limits: %v", err)
```
Comment (ncdc) [inline pkg/kubectl/describe.go]: Don't log this as the user shouldn't need to see it

**Comment by derekwaynecarr** [comment]
for limits that require introspection, i would just say "node capacity"

**Comment by ncdc** [comment]
@derekwaynecarr based on the discussion in #29639, should Avesh update this to just display "node allocatable"? (and stop trying to retrieve node info)

**Comment by derekwaynecarr** [comment]
LGTM

**Pull request merged**